railties/lib/rails/application/configuration.rb

```diff
@@ -514,7 +514,7 @@ def secret_key_base
       end
 
       def secret_key_base=(new_secret_key_base)
-        if new_secret_key_base.nil? && generate_local_secret?
+        if new_secret_key_base.nil? && Rails.env.local?
```

Comment:
Does this still work if an app does `secret_key_base = nil` and has `ENV["SECRET_KEY_BASE_DUMMY"]` set?

Reply:
hmm, yes probably not. There isn't a test case for that now. I'll try to add it.